src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java

```diff
@@ -161,17 +162,25 @@ public ErrorHandling getErrorHandling() {
                                                             SalesforceSinkConfig.PROPERTY_ERROR_HANDLING));
   }
 
-  public void validate(Schema schema) {
-    super.validate();
+  public void validate(Schema schema, FailureCollector collector) {
+    super.validate(collector);
 
     if (!containsMacro(PROPERTY_ERROR_HANDLING)) {
       // triggering getter will also trigger value validity check
-      getErrorHandling();
+      try {
+        getErrorHandling();
+      } catch (InvalidConfigPropertyException e) {
```

Comment:
since this is a deprecated class, it's better to remove usages of it, either by passing in the FailureCollector to getErrorHandling, or using a different type of exception.

Reply:
Done replaced with `InvalidConfigException`